test/support/coverage.js

```diff
@@ -21,16 +21,22 @@ if (Cypress.env('COVERAGE')) {
     cy.window({ log: false }).then(storeCoverage);
   });
 
+
   afterEach(function() {
     each(uniq(windowCoverageObjects), coverage => {
       cy.task('coverage', coverage);
     });
 
-    cy.task('write');
+    cy.task('coverage', global.__coverage__);
+
+    cy.window().then(win => {
```

Comment:
there are some circumstances where adding the coverage at this moment fixes the report issues.  I suspect this is somehow related to adding the report async slightly too early., but I can't quite put my finger on the issue.